File: Data Structure I/java/Day 3/BestTimeToBuyShellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        /**
        To pointer approach, similear to kadane's algo
         */
        int l=0;
        int r=1;
        int max=0;
        while(r<prices.length){
            if(prices[l]>=prices[r]){
                l=r;
            }else 
            if(prices[l]<prices[r]&&max<(prices[r]-prices[l])){
                max=prices[r]-prices[l];
            }
            r++;
        }
        return max;
    }
}

File: Data Structure I/python/Day 3/best_time_to_buy_shell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        '''
        To pointer approach, similear to kadane's algo
        '''
        l=0
        r=1
        max=0
        while(r<len(prices)):
            if(prices[l]>=prices[r]):
                l=r
            if(prices[l]<prices[r]  and max<(prices[r]-prices[l])):
                max=prices[r]-prices[l]
            r+=1
        return max